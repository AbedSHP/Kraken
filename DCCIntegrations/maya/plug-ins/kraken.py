import sys, json

from PySide import QtGui

import kraken.ui
from kraken.ui.kraken_ui import KrakenUI

from maya import cmds
# import maya.OpenMaya as OpenMaya
import maya.OpenMayaUI as OpenMayaUI
import maya.OpenMayaMPx as OpenMayaMPx

try:
  # Maya 2013 with custom pyside build
  import PySide.shiboken as shiboken
except:
  # Maya 2014 and higher
  import shiboken



# Command
class OpenKrakenEditorCommand(OpenMayaMPx.MPxCommand):
  def __init__(self):
    OpenMayaMPx.MPxCommand.__init__(self)

  # Invoked when the command is run.
  def doIt(self,argList):
    parent = OpenMayaUI.MQtUtil.mainWindow()
    parent = shiboken.wrapInstance(long(parent), QtGui.QWidget)
    dfgApp = KrakenUI(parent)

  # Creator
  @staticmethod
  def creator():
    return OpenMayaMPx.asMPxPtr( OpenKrakenEditorCommand() )


class KrakenUndoableCmd(OpenMayaMPx.MPxCommand):

  def __init__(self):
    OpenMayaMPx.MPxCommand.__init__(self)

  def isUndoable(self):
    return True

  def doIt(self, argList):
    return 0

  def redoIt(self):
    SingletonHost.getController().redo()
    return 0

  def undoIt(self):
    SingletonHost.getController().undo()
    return 0

  @staticmethod
  def creator():
    return OpenMayaMPx.asMPxPtr( KrakenUndoableCmd() )


# Initialize the script plug-in
def initializePlugin(mobject):
  mplugin = OpenMayaMPx.MFnPlugin(mobject)

  try:
    mplugin.registerCommand( 'openKrakenEditor', OpenKrakenEditorCommand.creator )
  except:
    sys.stderr.write( 'Failed to register DFG commands:openKrakenEditor' )
    raise

  try:
    mplugin.registerCommand( 'krakenUndoableCmd', KrakenUndoableCmd.creator )
  except:
    sys.stderr.write( 'Failed to register DFG commands:krakenUndoableCmd' )
    raise

# Uninitialize the script plug-in
def uninitializePlugin(mobject):
  mplugin = OpenMayaMPx.MFnPlugin(mobject)

  try:
    mplugin.deregisterCommand( 'openKrakenEditor' )
  except:
    sys.stderr.write( 'Failed to unregister command: openKrakenEditor' )


  try:
    mplugin.deregisterCommand( 'krakenUndoableCmd' )
  except:
    sys.stderr.write( 'Failed to unregister command: krakenUndoableCmd' )
