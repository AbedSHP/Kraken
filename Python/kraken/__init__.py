"""Kraken Framework."""

__all__ = ['core', 'examples', 'helpers', 'plugins', 'gui']