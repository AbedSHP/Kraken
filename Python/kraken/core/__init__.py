"""Kraken Core."""

VERSION_MAJOR = 0
VERSION_MINOR = 0
VERSION_BUILD = 1
VERSION_SUFFIX = "alpha"


def getVersion():
    """Contatenates the version globals and returns the current version of
    Kraken.

    Return:
    String, current version of Kraken.

    """

    versionString = str(VERSION_MAJOR) + "." + str(VERSION_MINOR) + "." + str(VERSION_BUILD)
    if VERSION_SUFFIX:
        versionString = versionString + "-" + VERSION_SUFFIX

    return versionString
