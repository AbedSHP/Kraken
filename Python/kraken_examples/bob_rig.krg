{
  "connections": [
    {
      "source": "spine:M.spineEnd",
      "target": "neck:M.neckBase"
    },
    {
      "source": "neck:M.neckEnd",
      "target": "head:M.headBase"
    },
    {
      "source": "spine:M.spineEnd",
      "target": "Clavicle:L.spineEnd"
    },
    {
      "source": "spine:M.spineEnd",
      "target": "Clavicle:R.spineEnd"
    },
    {
      "source": "Clavicle:L.clavicleEnd",
      "target": "Arm:L.clavicleEnd"
    },
    {
      "source": "Clavicle:R.clavicleEnd",
      "target": "Arm:R.clavicleEnd"
    },
    {
      "source": "spine:M.pelvis",
      "target": "Leg:L.pelvisInput"
    },
    {
      "source": "spine:M.pelvis",
      "target": "Leg:R.pelvisInput"
    }
  ],
  "graphPositions": {
    "head:M": {
      "y": 565.5626831054688,
      "x": -773.5648803710938,
      "__mathObjectClass__": "Vec2"
    },
    "neck:M": {
      "y": 551.1128540039062,
      "x": -982.2291259765625,
      "__mathObjectClass__": "Vec2"
    },
    "spine:M": {
      "y": 11.081206321716309,
      "x": -1383.1826171875,
      "__mathObjectClass__": "Vec2"
    },
    "Leg:R": {
      "y": -228.83909606933594,
      "x": -975.4631958007812,
      "__mathObjectClass__": "Vec2"
    },
    "Clavicle:R": {
      "y": 283.7149963378906,
      "x": -977.65185546875,
      "__mathObjectClass__": "Vec2"
    },
    "Clavicle:L": {
      "y": 21.802101135253906,
      "x": -971.165771484375,
      "__mathObjectClass__": "Vec2"
    },
    "Leg:L": {
      "y": -499.2738342285156,
      "x": -979.3851318359375,
      "__mathObjectClass__": "Vec2"
    },
    "Arm:L": {
      "y": 54.11977767944336,
      "x": -765.1942138671875,
      "__mathObjectClass__": "Vec2"
    },
    "Arm:R": {
      "y": 309.9880065917969,
      "x": -776.0560302734375,
      "__mathObjectClass__": "Vec2"
    }
  },
  "name": "char_bob",
  "components": [
    {
      "spine03Position": {
        "y": 12.449600219726562,
        "x": 0.0,
        "__mathObjectClass__": "Vec3",
        "z": -0.36489999294281006
      },
      "spine02Position": {
        "y": 11.801300048828125,
        "x": 0.0,
        "__mathObjectClass__": "Vec3",
        "z": -0.19949999451637268
      },
      "name": "spine",
      "numDeformers": 6,
      "spine01Position": {
        "y": 11.135100364685059,
        "x": 0.0,
        "__mathObjectClass__": "Vec3",
        "z": -0.13819999992847443
      },
      "cogPosition": {
        "y": 11.135100364685059,
        "x": 0.0,
        "__mathObjectClass__": "Vec3",
        "z": -0.13819999992847443
      },
      "spine04Position": {
        "y": 13.10509967803955,
        "x": 0.0,
        "__mathObjectClass__": "Vec3",
        "z": -0.4821000099182129
      },
      "class": "kraken_examples.spine_component.SpineComponentGuide",
      "location": "M"
    },
    {
      "location": "M",
      "neckPosition": {
        "y": 16.557199478149414,
        "x": 0.0,
        "__mathObjectClass__": "Vec3",
        "z": -0.6915000081062317
      },
      "class": "kraken_examples.neck_component.NeckComponentGuide",
      "neckEndPosition": {
        "y": 17.47559928894043,
        "x": 0.0,
        "__mathObjectClass__": "Vec3",
        "z": -0.42100000381469727
      },
      "name": "neck"
    },
    {
      "headPosition": {
        "y": 17.47559928894043,
        "x": 0.0,
        "__mathObjectClass__": "Vec3",
        "z": -0.42100000381469727
      },
      "name": "head",
      "jawPosition": {
        "y": 17.613000869750977,
        "x": 0.0,
        "__mathObjectClass__": "Vec3",
        "z": -0.27309998869895935
      },
      "eyeRightPosition": {
        "y": 18.087799072265625,
        "x": -0.349700003862381,
        "__mathObjectClass__": "Vec3",
        "z": 0.6087999939918518
      },
      "headEndPosition": {
        "y": 19.5,
        "x": 0.0,
        "__mathObjectClass__": "Vec3",
        "z": -0.42100000381469727
      },
      "class": "kraken_examples.head_component.HeadComponentGuide",
      "eyeLeftPosition": {
        "y": 18.087799072265625,
        "x": 0.349700003862381,
        "__mathObjectClass__": "Vec3",
        "z": 0.6087999939918518
      },
      "location": "M"
    },
    {
      "name": "Clavicle",
      "clavicleUpVXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 31.805999755859375,
          "x": 0.26440000534057617,
          "__mathObjectClass__": "Vec3",
          "z": -1.1446000337600708
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "clavicleEndXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 15.295000076293945,
          "x": 2.2699999809265137,
          "__mathObjectClass__": "Vec3",
          "z": -0.753000020980835
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "location": "L",
      "class": "kraken_examples.clavicle_component.ClavicleComponentGuide",
      "clavicleXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 15.402999877929688,
          "x": 0.13220000267028809,
          "__mathObjectClass__": "Vec3",
          "z": -0.5723000168800354
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      }
    },
    {
      "name": "Clavicle",
      "clavicleUpVXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 31.805999755859375,
          "x": -0.26440000534057617,
          "__mathObjectClass__": "Vec3",
          "z": -1.1446000337600708
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "clavicleEndXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 15.295000076293945,
          "x": -2.2699999809265137,
          "__mathObjectClass__": "Vec3",
          "z": -0.753000020980835
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "location": "R",
      "class": "kraken_examples.clavicle_component.ClavicleComponentGuide",
      "clavicleXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 15.402999877929688,
          "x": -0.13220000267028809,
          "__mathObjectClass__": "Vec3",
          "z": -0.5723000168800354
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      }
    },
    {
      "forearmFKCtrlSize": 1.5,
      "wristXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 12.281900405883789,
          "x": 7.188600063323975,
          "__mathObjectClass__": "Vec3",
          "z": 0.49059998989105225
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "name": "Arm",
      "bicepXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 15.295000076293945,
          "x": 2.2699999809265137,
          "__mathObjectClass__": "Vec3",
          "z": -0.753000020980835
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "bicepFKCtrlSize": 1.75,
      "forearmXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 13.5600004196167,
          "x": 5.039000034332275,
          "__mathObjectClass__": "Vec3",
          "z": -0.859000027179718
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "class": "kraken_examples.arm_component.ArmComponentGuide",
      "handXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 12.281900405883789,
          "x": 7.188600063323975,
          "__mathObjectClass__": "Vec3",
          "z": 0.49059998989105225
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 0.9330999851226807,
          "v": {
            "y": -0.23010000586509705,
            "x": -0.08649999648332596,
            "__mathObjectClass__": "Vec3",
            "z": -0.2623000144958496
          }
        }
      },
      "location": "L"
    },
    {
      "forearmFKCtrlSize": 1.5,
      "wristXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 12.281900405883789,
          "x": -7.188600063323975,
          "__mathObjectClass__": "Vec3",
          "z": 0.49059998989105225
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "name": "Arm",
      "bicepXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 15.295000076293945,
          "x": -2.2699999809265137,
          "__mathObjectClass__": "Vec3",
          "z": -0.753000020980835
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "bicepFKCtrlSize": 1.75,
      "forearmXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 13.5600004196167,
          "x": -5.039000034332275,
          "__mathObjectClass__": "Vec3",
          "z": -0.859000027179718
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "class": "kraken_examples.arm_component.ArmComponentGuide",
      "handXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 12.281900405883789,
          "x": -7.188600063323975,
          "__mathObjectClass__": "Vec3",
          "z": 0.49059998989105225
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 0.2623000144958496,
          "v": {
            "y": -0.08649999648332596,
            "x": -0.23010000586509705,
            "__mathObjectClass__": "Vec3",
            "z": -0.9330999851226807
          }
        }
      },
      "location": "R"
    },
    {
      "toeTipXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 0.4000000059604645,
          "x": 1.850000023841858,
          "__mathObjectClass__": "Vec3",
          "z": 1.5
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "name": "Leg",
      "ankleXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 1.1516000032424927,
          "x": 1.840999960899353,
          "__mathObjectClass__": "Vec3",
          "z": -1.2369999885559082
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "kneeXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 5.441800117492676,
          "x": 1.448799967765808,
          "__mathObjectClass__": "Vec3",
          "z": -0.5347999930381775
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "toeXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 0.4000000059604645,
          "x": 1.850000023841858,
          "__mathObjectClass__": "Vec3",
          "z": 0.25
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "class": "kraken_examples.leg_component.LegComponentGuide",
      "femurXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 9.769000053405762,
          "x": 0.9811000227928162,
          "__mathObjectClass__": "Vec3",
          "z": -0.45719999074935913
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "location": "L"
    },
    {
      "toeTipXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 0.4000000059604645,
          "x": -1.850000023841858,
          "__mathObjectClass__": "Vec3",
          "z": 1.5
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "name": "Leg",
      "ankleXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 1.1516000032424927,
          "x": -1.840999960899353,
          "__mathObjectClass__": "Vec3",
          "z": -1.2369999885559082
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "kneeXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 5.441800117492676,
          "x": -1.448799967765808,
          "__mathObjectClass__": "Vec3",
          "z": -0.5347999930381775
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "toeXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 0.4000000059604645,
          "x": -1.850000023841858,
          "__mathObjectClass__": "Vec3",
          "z": 0.25
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "class": "kraken_examples.leg_component.LegComponentGuide",
      "femurXfo": {
        "sc": {
          "y": 1.0,
          "x": 1.0,
          "__mathObjectClass__": "Vec3",
          "z": 1.0
        },
        "__mathObjectClass__": "Xfo",
        "tr": {
          "y": 9.769000053405762,
          "x": -0.9811000227928162,
          "__mathObjectClass__": "Vec3",
          "z": -0.45719999074935913
        },
        "ori": {
          "__mathObjectClass__": "Quat",
          "w": 1.0,
          "v": {
            "y": 0.0,
            "x": 0.0,
            "__mathObjectClass__": "Vec3",
            "z": 0.0
          }
        }
      },
      "location": "R"
    }
  ]
}