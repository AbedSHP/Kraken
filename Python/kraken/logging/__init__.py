
import logging

from kraken.plugins import getLogHandler


def getLogger(name):
    """Get's a logger and attaches the correct DCC compatible Handler.

    Args:
        name (str): Name of the logger to get / create.

    Returns:
        Logger: Logger.

    """

    dccHandler = getLogHandler()
    logger = logging.getLogger(name)
    logger.addHandler(dccHandler)

    return logger
