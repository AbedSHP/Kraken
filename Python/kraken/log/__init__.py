
import logging

from kraken.plugins import getLogHandler


def getLogger(name):
    """Get's a logger and attaches the correct DCC compatible Handler.

    Args:
        name (str): Name of the logger to get / create.

    Returns:
        Logger: Logger.

    """

    dccHandler = getLogHandler()
    logger = logging.getLogger(name)
    if dccHandler not in logger.handlers:
        logger.addHandler(dccHandler)

    return logger
