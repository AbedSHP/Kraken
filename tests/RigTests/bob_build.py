from kraken import plugins
from kraken.tests.RigTests.bob_rig import Rig

bobRig = Rig("char_bob")

builder = plugins.getBuilder()
builder.build(bobRig)