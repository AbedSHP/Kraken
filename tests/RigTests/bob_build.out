Failed to find DCC builder. Falling back to Python builder.
char_bob
char_bob.deformers
char_bob.deformers.spine01
char_bob.deformers.spine02
char_bob.deformers.spine03
char_bob.deformers.spine04
char_bob.deformers.neck
char_bob.deformers.head
char_bob.deformers.jaw
char_bob.deformers.eyeLeft
char_bob.deformers.eyeRight
char_bob.deformers.clavicle
char_bob.deformers.clavicle
char_bob.deformers.bicep
char_bob.deformers.forearm
char_bob.deformers.wrist
char_bob.deformers.bicep
char_bob.deformers.forearm
char_bob.deformers.wrist
char_bob.deformers.hand
char_bob.deformers.hand
char_bob.deformers.femur
char_bob.deformers.shin
char_bob.deformers.ankle
char_bob.deformers.femur
char_bob.deformers.shin
char_bob.deformers.ankle
char_bob.deformers.foot
char_bob.deformers.foot
char_bob.controls
char_bob.controls.spine
char_bob.controls.spine.inputs
char_bob.controls.spine.outputs
char_bob.controls.spine.outputs.spine00
char_bob.controls.spine.outputs.spine01
char_bob.controls.spine.outputs.spine02
char_bob.controls.spine.outputs.spine03
char_bob.controls.spine.outputs.spineBase
char_bob.controls.spine.outputs.spineEnd
char_bob.controls.spine.cog
char_bob.controls.spine.cog.cog
char_bob.controls.spine.cog.cog.spine01
char_bob.controls.spine.cog.cog.spine01.spine01
char_bob.controls.spine.cog.cog.spine01.spine01.spine02
char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02
char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03
char_bob.controls.spine.cog.cog.spine01.spine01.spine02.spine02.spine03.spine03
char_bob.controls.spine.cog.cog.spine04
char_bob.controls.spine.cog.cog.spine04.spine04
char_bob.controls.neck
char_bob.controls.neck.inputs
char_bob.controls.neck.inputs.neckBase
char_bob.controls.neck.outputs
char_bob.controls.neck.outputs.neckEnd
char_bob.controls.neck.outputs.neck
char_bob.controls.neck.neck
char_bob.controls.neck.neck.neck
char_bob.controls.head
char_bob.controls.head.inputs
char_bob.controls.head.inputs.headBase
char_bob.controls.head.outputs
char_bob.controls.head.outputs.head
char_bob.controls.head.outputs.jaw
char_bob.controls.head.outputs.eyeL
char_bob.controls.head.outputs.eyeR
char_bob.controls.head.head
char_bob.controls.head.head.head
char_bob.controls.head.head.head.eyeLeft
char_bob.controls.head.head.head.eyeLeft.eyeLeft
char_bob.controls.head.head.head.eyeRight
char_bob.controls.head.head.head.eyeRight.eyeRight
char_bob.controls.head.head.head.jawCtrlSpace
char_bob.controls.head.head.head.jawCtrlSpace.jaw
char_bob.controls.clavicle
char_bob.controls.clavicle.inputs
char_bob.controls.clavicle.inputs.spineEnd
char_bob.controls.clavicle.outputs
char_bob.controls.clavicle.outputs.clavicleEnd
char_bob.controls.clavicle.outputs.clavicle
char_bob.controls.clavicle.clavicle
char_bob.controls.clavicle.clavicle.clavicle
char_bob.controls.clavicle
char_bob.controls.clavicle.inputs
char_bob.controls.clavicle.inputs.spineEnd
char_bob.controls.clavicle.outputs
char_bob.controls.clavicle.outputs.clavicleEnd
char_bob.controls.clavicle.outputs.clavicle
char_bob.controls.clavicle.clavicle
char_bob.controls.clavicle.clavicle.clavicle
char_bob.controls.arm
char_bob.controls.arm.inputs
char_bob.controls.arm.inputs.clavicleEnd
char_bob.controls.arm.outputs
char_bob.controls.arm.outputs.bicep
char_bob.controls.arm.outputs.forearm
char_bob.controls.arm.outputs.armEndXfo
char_bob.controls.arm.outputs.armEndPos
char_bob.controls.arm.bicepFK
char_bob.controls.arm.bicepFK.bicepFK
char_bob.controls.arm.bicepFK.bicepFK.forearmFK
char_bob.controls.arm.bicepFK.bicepFK.forearmFK.forearmFK
char_bob.controls.arm.IK
char_bob.controls.arm.IK.IK
char_bob.controls.arm.UpV
char_bob.controls.arm.UpV.UpV
char_bob.controls.arm
char_bob.controls.arm.inputs
char_bob.controls.arm.inputs.clavicleEnd
char_bob.controls.arm.outputs
char_bob.controls.arm.outputs.bicep
char_bob.controls.arm.outputs.forearm
char_bob.controls.arm.outputs.armEndXfo
char_bob.controls.arm.outputs.armEndPos
char_bob.controls.arm.bicepFK
char_bob.controls.arm.bicepFK.bicepFK
char_bob.controls.arm.bicepFK.bicepFK.forearmFK
char_bob.controls.arm.bicepFK.bicepFK.forearmFK.forearmFK
char_bob.controls.arm.IK
char_bob.controls.arm.IK.IK
char_bob.controls.arm.UpV
char_bob.controls.arm.UpV.UpV
char_bob.controls.hand
char_bob.controls.hand.inputs
char_bob.controls.hand.inputs.armEndXfo
char_bob.controls.hand.inputs.armEndPos
char_bob.controls.hand.outputs
char_bob.controls.hand.outputs.hand
char_bob.controls.hand.outputs.handEnd
char_bob.controls.hand.hand
char_bob.controls.hand.hand.hand
char_bob.controls.hand.handRef
char_bob.controls.hand
char_bob.controls.hand.inputs
char_bob.controls.hand.inputs.armEndXfo
char_bob.controls.hand.inputs.armEndPos
char_bob.controls.hand.outputs
char_bob.controls.hand.outputs.hand
char_bob.controls.hand.outputs.handEnd
char_bob.controls.hand.hand
char_bob.controls.hand.hand.hand
char_bob.controls.hand.handRef
char_bob.controls.leg
char_bob.controls.leg.inputs
char_bob.controls.leg.inputs.pelvisInput
char_bob.controls.leg.outputs
char_bob.controls.leg.outputs.femur
char_bob.controls.leg.outputs.shin
char_bob.controls.leg.outputs.legEndXfo
char_bob.controls.leg.outputs.legEndPos
char_bob.controls.leg.femurFK
char_bob.controls.leg.femurFK.femurFK
char_bob.controls.leg.femurFK.femurFK.shinFK
char_bob.controls.leg.femurFK.femurFK.shinFK.shinFK
char_bob.controls.leg.IK
char_bob.controls.leg.IK.IK
char_bob.controls.leg.UpV
char_bob.controls.leg.UpV.UpV
char_bob.controls.leg
char_bob.controls.leg.inputs
char_bob.controls.leg.inputs.pelvisInput
char_bob.controls.leg.outputs
char_bob.controls.leg.outputs.femur
char_bob.controls.leg.outputs.shin
char_bob.controls.leg.outputs.legEndXfo
char_bob.controls.leg.outputs.legEndPos
char_bob.controls.leg.femurFK
char_bob.controls.leg.femurFK.femurFK
char_bob.controls.leg.femurFK.femurFK.shinFK
char_bob.controls.leg.femurFK.femurFK.shinFK.shinFK
char_bob.controls.leg.IK
char_bob.controls.leg.IK.IK
char_bob.controls.leg.UpV
char_bob.controls.leg.UpV.UpV
char_bob.controls.foot
char_bob.controls.foot.inputs
char_bob.controls.foot.inputs.legEndXfo
char_bob.controls.foot.inputs.legEndPos
char_bob.controls.foot.outputs
char_bob.controls.foot.outputs.hand
char_bob.controls.foot.outputs.handEnd
char_bob.controls.foot.foot
char_bob.controls.foot.foot.foot
char_bob.controls.foot.footRef
char_bob.controls.foot
char_bob.controls.foot.inputs
char_bob.controls.foot.inputs.legEndXfo
char_bob.controls.foot.inputs.legEndPos
char_bob.controls.foot.outputs
char_bob.controls.foot.outputs.hand
char_bob.controls.foot.outputs.handEnd
char_bob.controls.foot.foot
char_bob.controls.foot.foot.foot
char_bob.controls.foot.footRef
char_bob.geometry
